src/mesh/coordinates.cxx

```diff
@@ -505,6 +505,43 @@ int Coordinates::jacobian() {
   return 0;
 }
 
+/** This function calculates the L2 vector norm of the Linf matrix norm of gij*g_ij - I.
+ * 	This function can be used to test the consistency of metric the co- and contravariant tensors
+ **/
+BoutReal Coordinates::gijXg_ijMinusI(){
+	BoutReal globalError = 0.;
+	BoutReal localError = 0.;
+	///2d array of field3d pointers. Using array because vector of const references aor const pointers is not allowed
+	Field2D * const gij[3][3] = {{&g11,&g12,&g13}, {&g12,&g22,&g23}, {&g13,&g23,&g33}};
+	Field2D * const g_ij[3][3]= {{&g_11,&g_12,&g_13}, {&g_12,&g_22,&g_23}, {&g_13,&g_23,&g_33}};
+
+	BoutReal res[3][3] = {{0.,0.,0.},{0.,0.,0.},{0.,0.,0.}};
+	const BoutReal I[3][3] = {{1.,0.,0.},{0.,1.,0.},{0.,1.,0.}};
+
+	for(int x=mesh->xstart;x<mesh->xend;x++)for(int y=mesh->ystart;y<mesh->yend;y++)
```

Comment:
Isn't `mesh->xend` still within the grid?
I think this is where we deviate from the C++ end definition ...